### PR TITLE
WT-6356 Assert we don't see out of order timestamps

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Comment by bvpvamsikrishna** [comment]
@quchenhao , could you trigger a patch build for this PR. My preference is, not to get this PR in for today's release train, so I will review this in the later part of the day.

**Comment by quchenhao** [comment]
@bvpvamsikrishna patch build created
https://evergreen.mongodb.com/version/5ed6ed980ae606496cf4d1f1

File: src/history/hs.c [hunk 1]
```diff
@@ -481,17 +481,22 @@ __hs_insert_record_with_btree(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BT
     WT_ASSERT(session, type == WT_UPDATE_STANDARD || type == WT_UPDATE_MODIFY);
 
     /*
-     * If the time points are out of order (which can happen if the application performs updates
-     * with out-of-order timestamps), so this value can never be seen, don't bother inserting it.
+     * If the application performs updates with mixed mode transactions, this value can never be
```
Comment (luke-pearson) [inline src/history/hs.c:484]: I think mixed mode timestamps is a better phrasing, and perhaps an explanation of what "mixed mode" timestamps are would be helpful in this comment, i.e. having a zero timestamp which effectively makes the rest of the chain obsolete. 

Mixed mode transactions makes me think we're running at different isolation levels or something.

**Comment by quchenhao** [comment]
Close as we need to support out of order timestamp for now.

**Pull request closed without merging**